### ISSUE TITLE
fitting Mbol & WD mass

### ISSUE THREAD

**Issue by sparx00** [cylammarco/WDPhotTools#24]
Hi, I am trying to fit for the WD mass and Mbol. Broadly, I am following the approach outlined here: https://wdphottools.readthedocs.io/en/latest/tutorials/fitting.html. From what I understood, one needs to pick for the independent list one parameter from (Mbol or Teff) and the second one from (mass/Mbol) for the fitting.  I am trying to run something like this:

```
from WDPhotTools.fitter import WDfitter
ftr = WDfitter()

ftr.fit(
    atmosphere="H",
    filters=["G3", "G3_BP", "G3_RP"],
    mags=[19.1, 18.8, 19.2],
    mag_errors=[0.006, 0.01, 0.07],
    distance=210.0,
    distance_err=10,
    extinction_convolved=True,
    kind="cubic",
    Rv=3.1,
    ebv=0.1,
    independent=["Mbol", "mass"],
    initial_guess=[13.0, 0.6],
    method="least_squares"
   )
```

but I keep getting this error:

```
ValueError                                Traceback (most recent call last)

[<ipython-input-58-2b54e9b12163>](https://localhost:8080/#) in <cell line: 4>()
      2 ftr = WDfitter()
      3 
----> 4 ftr.fit(
      5     atmosphere="H",
      6     filters=["G3", "G3_BP", "G3_RP"],

[/usr/local/lib/python3.9/dist-packages/WDPhotTools/fitter.py](https://localhost:8080/#) in fit(self, atmosphere, filters, mags, mag_errors, allow_none, distance, distance_err, extinction_convolved, kind, Rv, ebv, ra, dec, independent, initial_guess, logg, atmosphere_interpolator, reuse_interpolator, method, nwalkers, nsteps, nburns, progress, refine, refine_bounds, kwargs_for_RBF, kwargs_for_CT, kwargs_for_minimize, kwargs_for_least_squares, kwargs_for_emcee)
   2061                         # the [:2] is to separate the distance from the filters
   2062                         self.best_fit_params[j][i] = float(
-> 2063                             self.interpolator[j][i](self.results[j].x[0])
   2064                         )
   2065 

[/usr/local/lib/python3.9/dist-packages/WDPhotTools/atmosphere_model_reader.py](https://localhost:8080/#) in atmosphere_interpolator(*x)
    486                 def atmosphere_interpolator(*x):
    487 
--> 488                     x_0, x_1 = np.asarray(x, dtype="object").reshape(-1)
    489 
    490                     if independent[0] in ["Teff", "age"]:

ValueError: not enough values to unpack (expected 2, got 1)
```

I am probably missing something obvious. I would be glad for any help.
Thanks.

**Comment by cylammarco** [comment]
Still working on the decision tree of the possible combinations of `logg` and `mass`.

**Comment by cylammarco** [comment]
Hi @sparx00, I would like to consult your opinion on this. It seems to me that it makes the most sense to allow fitting of 2 variables that are selected from 2 groups:

1. Temperature, Mbol (/Luminosity), Age
2. Mass, logg

Given WDPhotTools does not have the option the change the mass-radius relation, do you see a need of fixing logg and then fit for the mass and a parameter from group 1?

**Comment by cylammarco** [comment]
This bug is fixed in v0.0.8.

**Comment by sparx00** [comment]
Hi, thank you for your efforts, and sorry for the delayed response.

> It seems to me that it makes the most sense to allow fitting of 2 variables that are selected from 2 groups:
> 1. Temperature, Mbol (/Luminosity), Age
> 2. Mass, logg

Yes, that also makes sense to me.

> Given WDPhotTools does not have the option the change the mass-radius relation, do you see a need of fixing logg and then fit for the mass and a parameter from group 1?

I am not familiar with all possible usecases, but I have not run into a case where it would be required.
